tests/integration/fixtures.py

```diff
@@ -169,12 +169,13 @@ def gpu_string_table(string_table):
 
 @pytest.fixture()
 def gpu_datetime_table(datetime_table):
-    # cudf doesn't have support for timezoned datetime data
-    datetime_table["timezone"] = datetime_table["timezone"].astype("datetime64[ns]")
-    datetime_table["utc_timezone"] = datetime_table["utc_timezone"].astype(
-        "datetime64[ns]"
-    )
-    return cudf.from_pandas(datetime_table) if cudf else None
+    if cudf:
+        # cudf doesn't have support for timezoned datetime data
+        df = datetime_table.copy()
+        df["timezone"] = df["timezone"].dt.tz_localize(None)
+        df["utc_timezone"] = df["utc_timezone"].dt.tz_localize(None)
+        return cudf.from_pandas(df)
+    return None
```

Comment:
Before this change, we were actually overwriting the `datetime_table` CPU fixture such that it didn't contain timezone-aware data.

Now that we're no longer doing that, a failure has been exposed in `test_filter_cast_timestamp` because filtering a timezone-aware column by a timestamp would require non-UTC timezone handling for literals on the Python end:

https://github.com/dask-contrib/dask-sql/blob/899170679b44b8eebd5efe1af373eded98cbe9aa/dask_sql/physical/rex/core/literal.py#L181-L182

There are a few ways we could handle this, but all of them generally involve front-facing changes to timestamp literal handling so figure it might make sense to break that off into a separate PR.

Okay with xfailing that test for now to get this in? cc @ayushdg @jdye64